docs/rfcs/0004-rfc-connectiomanager.md

```diff
@@ -0,0 +1,76 @@
+- Status: In progress
+- Date: 2023-11-02
+- Author: [@gela](https://github.com/gela)
+- RFC PR: TBD
+- Implementation GitHub issue: TBD.
+
+
+## Summary
+
+Propose a new `ConnectionManager` component which handles all connectivity related checks once during VPN activation (`Controller::StateConnecting`) and then periodically once the VPN is already activated (`Controller::StateOn`); the controller will continue to exist as is, communicating with the frontend as the backend goes through various states such as `connecting` and `confirming` to reflect those to the user.
+
+## Motivation
+
+At present, the handling of VPN connectivity lacks standardization and consistency. There are instances when the VPN goes into No Signal due to a variety of connectivity-related issues, such as the network connection failing or the server location becoming unavailable. In these situations, users are often left without a clear explanation of what went wrong or guidance on how to troubleshoot and resolve the issue. They may encounter a background error message or lose signal without further information.
```

Comment:
"lacks standardization and consistency" - You mean for users, right? Not in the code? That's what the rest of the paragraph implies, but it's probably worth being explicit.